reconcile/openshift_saas_deploy_trigger_base.py

```diff
@@ -321,6 +321,22 @@ def _pipeline_exists(
     return False
 
 
+def _build_pipeline_run_long_name(
+    saas_file_name: str,
+    env_name: str,
+) -> str:
+    return f"{saas_file_name}-{env_name}".lower()
+
+
+def build_pipeline_run_name_prefix(
+    saas_file_name: str,
+    env_name: str,
+) -> str:
+    return _build_pipeline_run_long_name(saas_file_name, env_name)[
+        :UNIQUE_SAAS_FILE_ENV_COMBO_LEN
+    ]
```

Comment:
nit: would it make sense to have a small comment here that uniqueness for `UNIQUE_SAAS_FILE_ENV_COMBO_LEN` is already ensured by saasherder? It might also be redundant to add that comment, as the constant naming already might suggest it. But still I was checking now if uniqueness is actually guaranteed here :sweat_smile:

Reply:
I actually have no idea what this constant means, just extracting existing logic to this func...

But you pointed out another usage of `UNIQUE_SAAS_FILE_ENV_COMBO_LEN`, let me see if I can reuse this logic in one more place, so this `UNIQUE_SAAS_FILE_ENV_COMBO_LEN` is not needed here anymore.

Reply:
@fishi0x01 I extracted another function from [saasherder.py](https://github.com/app-sre/qontract-reconcile/pull/3559/files#diff-52a815feb4bcd8276212403b5f0ae22e3a3d9508acb5b7a75a5134dcf2657bbcR473), more strict than before (case insensitive by `lower()`), should be ok, right?

Reply:
I honestly find no reason that it shouldn't work, but saasherder.py is full of surprises. I would refrain from changing stuff beyond the nice refactor and I would remove the lower 😅

Reply:
@rporres that's a good point! It's already not consistent, in saas herder, the unique check is using case sensitive (no lower call), in actual saas deploy trigger run, it's using case insensitive (call lower). As it's not related to the issue I'm trying to fix, I updated the logic to keep the original behaviour, only reuse it to build console url, which case doesn't matter in UI search. Please have another look, thanks!

Reply:
It looks great. Thx!